September news (#642)
ropensci/roweb3: content/blog/2023-09-22-news-september-2023/index.Rmd

description: Open Call for Champions program, community call in Spanish and English, coworking
params:
  last_newsletter: "2023-08-25"
---

```{r setup, include=FALSE}
library("magrittr")
library("rlang")
last_newsletter <- anytime::anytime(params$last_newsletter)
knitr::opts_chunk$set(echo = FALSE)
url <- sprintf(
    "/blog/%s/%s/%s/%s",
    lubridate::year(rmarkdown::yaml_front_matter(knitr::current_input())$date),
    stringr::str_pad(lubridate::month(rmarkdown::yaml_front_matter(knitr::current_input())$date), 2, "0", side = "left"),
    stringr::str_pad(lubridate::day(rmarkdown::yaml_front_matter(knitr::current_input())$date), 2, "0", side = "left"),
    rmarkdown::yaml_front_matter(knitr::current_input())$slug
    )
english <- function(x) {
  as.character(english::english(x))
}

nice_string <- function(...) {
  glue::glue_collapse(..., sep = ", ", last = ", and ")
}
```
<!-- Before sending DELETE THE INDEX_CACHE and re-knit! -->

Dear rOpenSci friends, it's time for our monthly news roundup!
<!-- blabla -->
You can read this post [on our blog](`r url`).
Now let's dive into the activity at and around rOpenSci!

## rOpenSci HQ

### WIP: WebAssembly support in R-universe!

> Thanks to some help from George Stagg, we added experimental support for building WebAssembly binary packages. This makes it possible to install packages in [webr](https://docs.r-wasm.org/webr/latest/), directly from r-universe.

Reviewer: ```suggestion
Thanks to some help from George Stagg, we added experimental support for building WebAssembly binary packages. This makes it possible to install packages in [webr](https://docs.r-wasm.org/webr/latest/), directly from the R-universe.
```

For instance: 

```r
webr::install('jsonlite', repos = c('https://jeroen.r-universe.dev', 'https://repo.r-wasm.org'))
```

This only works if the package and all of its dependencies support webassembly.

> For CRAN packages you can look at https://repo.r-wasm.org/; for other packages... you have to try.

Reviewer: ```suggestion
For CRAN packages you can look at https://repo.r-wasm.org/; for other packages... you'll have to give it a try!
```

> ### First champion package to go through peer-review, in Spanish!

Reviewer: ```suggestion
### First rOpenSci Champion's package to go through peer-review, in Spanish!
```

[Carolina Pradier](/blog/2023/05/05/ropensci-champions-program-teams-meet-carolina-pradier-and-athanasia-monika-mowinckel/#carolina-pradier---champion)'s package [eph](https://docs.ropensci.org/eph/) was approved after a [review process in Spanish](https://github.com/ropensci/software-review/issues/593)!

> Congratulations to Carolina, and thanks to editor [Mauro Lepore](/author/mauro-lepore/), reviewers [Guadalupe Gonzalez](https://github.com/guadag12) and [Denisse Fierro Arcos](https://github.com/lidefi87), and mentor [Athanasia Monika Mowinckel](/blog/2023/05/05/ropensci-champions-program-teams-meet-carolina-pradier-and-athanasia-monika-mowinckel/#athanasia-monika-mowinckel---mentor).
> 
> eph helps processing data from the Argentina household survey.

Reviewer: ```suggestion
Congratulations to Carolina, and thanks to editor [Mauro Lepore](/author/mauro-lepore/), reviewers [Guadalupe Gonzalez](https://github.com/guadag12) and [Denisse Fierro Arcos](https://github.com/lidefi87), and mentor [Athanasia Monika Mowinckel](/blog/2023/05/05/ropensci-champions-program-teams-meet-carolina-pradier-and-athanasia-monika-mowinckel/#athanasia-monika-mowinckel---mentor). eph is a package which helps process data from the Argentina household survey.
```

### 123 Applications to our Champions Program 2023-2024

We are excited we received 123 applications from 41 countries on 5 continents!  
We are very grateful to everyone who submitted their proposals to our program.
The review process is starting. 
Stay tuned for updates!

### Coworking

Read [all about coworking](/blog/2023/06/21/coworking/) in our recent [post](/blog/2023/06/21/coworking/)!

Join us for social coworking & office hours monthly on first Tuesdays! 
Hosted by Steffi LaZerte and various community hosts. 
Everyone welcome. 
No RSVP needed. 
Consult our [Events](/events) page to find your local time and how to join.

> - [Tuesday, October 3rd, 14h00 Europe Central (12:00 UTC)](/events/coworking-2023-10/), Learn about cli and how to make nice command-line interfaces in R! With [community host Athanasia Monika Mowinckel](/author/athanasia-mo-mowinckel/) and [Steffi LaZerte](/author/steffi-lazerte/).
>     - Read up on making nice command line interfaces with [cli](https://cli.r-lib.org/);
>     - Update some scripts or packages to include cleaner and effective interfaces.

Reviewer: ```suggestion
- [Tuesday, October 3rd, 14h00 Europe Central (12:00 UTC)](/events/coworking-2023-10/), Learn about cli and how to make nice command-line interfaces in R! With community host [Athanasia Monika Mowinckel](/author/athanasia-mo-mowinckel/) and [Steffi LaZerte](/author/steffi-lazerte/).
    - Read up on making nice command line interfaces with [cli](https://cli.r-lib.org/);
    - Update some scripts or packages to include cleaner and effective interfaces;
    - Chat with Mo about using cli
```

> 

Reviewer: I will merge in the event PR before Friday so it should be ready by the time you publish 🤞🏼 

```suggestion
- [Tuesday, Nov 7th, 9:00 America Pacific (17:00 UTC)](/events/coworking-2023-11/), Fixing scary bugs 😱🐛 With community host [Salix Dubois](/author/salix-dubois/) and [Steffi LaZerte](/author/steffi-lazerte/).
    - Identify/fix some scary bugs;
    - Chat with Salix about scary bugs!

```

- **[natserv](https://cran.r-project.org/web/packages/natserv/index.html)**. Interface to NatureServe (<https://www.natureserve.org/>). Includes methods to get data, image metadata, search taxonomic names, and make maps. [Issue for volunteering](https://github.com/ropensci-archive/natserv/issues/29).

### Call for comaintainers

Refer to our somewhat [recent blog post](/blog/2022/10/17/maintain-or-co-maintain-an-ropensci-package/#packages-looking-for-co-maintainers) to identify other packages where help is especially wished for!
See also our [help wanted page](/help-wanted/) -- before opening a PR, we recommend asking in the issue whether help is still needed.

## Package development corner

Some useful tips for R package developers. :eyes:

### Help needed issues

Don't miss our blog post [Attract Contributors with ‘help wanted’ Issues](/blog/2023/09/19/help-wanted/)! :smile_cat:

### Check your contributing guide...

... with this [evaluation tool](https://contributing.streamlit.app/).

> ### Regularly check your pkgdown configuration file

Reviewer: Eeep, definitely something I should bear in mind!

Author: I do that mistake all the time myself :sob:
